### ISSUE TITLE
Fix project layout so dependencies on service_worker and googleapi_auth aren't visible

### ISSUE THREAD

**Issue by alextekartik** [FirebaseExtended/firebase-dart#270]
When building a web project using firebase using ddc, I get the following warning:
```
[WARNING]build_web_compilers:ddc_modules on package:firebase/.ddc.meta_module.raw: Unable to read module information for package:googleapis_auth, make sure you have a dependency on it in your pubspec.
[WARNING]build_web_compilers:ddc_modules on package:firebase/.ddc.meta_module.raw: Unable to read module information for package:service_worker, make sure you have a dependency on it in your pubspec.
```

Indeed some code in lib are importing these packages in src but the dependency are only dev_dependency. Is this normal? (Actually I would love to be able to expose a package with optional depencencies).

**Comment by kevmoo** [comment]
This is intentional, but weird. I put things in the lib directory I want to share between tests and examples.

These **should** be in a separate package, but that means I need to tweak the project a bunch.

**Comment by kevmoo** [comment]
Fixed in 7ef81823e9f0ab0995255d389e0af017e4771bd1
